### ISSUE TITLE
Issue with 'ensure_process = running' and group

### ISSUE THREAD

**Issue by Vincent--** [ajcrowe/puppet-supervisord#88]
I'm not sure the approach in this PR is working when using group...
https://github.com/ajcrowe/puppet-supervisord/pull/72

In my case, the program name is "ecard-generate-pdf_01" and the group name is "ecard".

```
root:/etc/supervisor/conf.d$ supervisorctl status
ecard:ecard-generate-pdf_01      STOPPED    Not started
```

If I try to start the program using only its name (as it's done by the puppet module), that doesn't work:
```
root:/etc/supervisor/conf.d$ supervisorctl start ecard-generate-pdf_01
ecard-generate-pdf_01: ERROR (no such process)
```

I need to use the full name of the program (ie, group:program):
```
root:/etc/supervisor/conf.d$ supervisorctl start ecard:ecard-generate-pdf_01
ecard:ecard-generate-pdf_01: started
```

Due to that, every time I run puppet, the module try to start the programs unsuccessfully....

**Comment by Vincent--** [comment]
Also, I don't want this module to start program with autostart = false...

**Comment by Vincent--** [comment]
Actually, I'm not sure we need the "ensure_process" logic... 

The same result can be obtained with autostart = yes and autorestart = 'true' (or even autorestart = 'unexpected'), IIUC

If we want to stop the program, we can use "autostart = no"
And if we don't need the program any more, we can just remove its configuration from puppet...

Can someone give more details about why we need the "stopped", "removed" and "running" values for ensure_process? Am I missing something?
In my opinion, supervisorctl should be used only for manual operations.

**Comment by ajcrowe** [comment]
I've merged #92 which should be an alternative to the current behaviour.

I think it would be good to make processes group aware, I will have to think about the best approach for that.
